l2geth/core/state_transition_ovm.go

```diff
@@ -29,7 +30,7 @@ func toExecutionManagerRun(evm *vm.EVM, msg Message) (Message, error) {
 		uint8(msg.QueueOrigin().Uint64()),
 		*msg.L1MessageSender(),
 		*msg.To(),
-		big.NewInt(int64(msg.Gas())),
+		new(big.Int).SetUint64(msg.Gas()),
```

Comment:
Is there a functional difference between these two approaches?

Reply:
Yes, `int64` will overflow at half of max `uint64` so this is actually currently broken. Prefer the `new(big.Int).SetUint64` syntax when dealing with `uint64`

Reply:
How is `msg.Gas()` set?  The correct behavior to get this to line up with fraud proofs should be:
- if this is an `enqueue`, the `tx.gasLimit` should be the argument to `enqueue`. (https://github.com/ethereum-optimism/optimism/blob/0c18e1903f8a33a607f782c0081a8fb5071b71ef/packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol#L1184)
- if this is a sequencer transaction, the `tx.gasLimit` should be the max OVM tx gas limit. (https://github.com/ethereum-optimism/optimism/blob/0c18e1903f8a33a607f782c0081a8fb5071b71ef/packages/contracts/contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol#L1123)
I suspect that this is broken for the latter case as it currently stands; I would assume that the `msg.Gas` gets populated by the user transaction right now.

Reply:
Ah @ben-chain it should actually currently supply the full gasLimit. I've proved this to myself a few times and can dig it up again, but unless something has changed our behavior is correct *I believe*. The only thing that _might_ be wrong is if we supply max gas limit to all deposits

Reply:
yeah, @kelvin directed me a little bit and it looks like it's being set to the `header.gasLimit` via [this](https://github.com/ethereum-optimism/optimism/blob/0c18e1903f8a33a607f782c0081a8fb5071b71ef/l2geth/core/state_processor.go#L93).  If that header is correctly set to the `MaxTransactionGasLimit`, then we are good for sequencer transactions, but it seems like we are probably supplying too much gas for deposits as you say.  That can be a separate PR I guess?

Reply:
https://github.com/ethereum-optimism/optimism/blob/4d2a7d5f25e215c420fa26e2fa57958cf396936c/l2geth/core/state_transition_ovm.go#L52

Reply:
I think this is appropriate for another PR. It would be nice to have a spec for these values along with a refactor to the minimal diff